### ISSUE TITLE
Literate element search

### ISSUE THREAD

**Issue by NathanZook** [microsoft/WinAppDriver#889]
Digging into this, I find myself wanting a more "literate" way of searching for elements.  My idea is something like this:

```
session.
      ChildPane("My Application").
      ChildDocument("Question for you").
      WithAttribute("HasKeyBoardFocus", true).
      DescendantText("What I want selected").
      Get();
```

This spares user from having to figure out the corner cases of XPath support.  For usual cases, it translates directly into XPath calls.  For weirdness like dialogs, this is hidden from the user.  Bugs in a particular method can be isolated.  Superpowerful, however is the final `Get()`.  Of course, `GetAll()` is an obvious complement, but also, we can do `WaitFor(Timespan)`, or just extend the thing with whatever complex logic works for our test case.

**Comment by KitoCoding** [comment]
I disagree, this method forces recursive searching for an element based on your criteria. Unless you are proposing a somewhat of an 'XPath' builder, which all this is doing is building a final string which we search off in the end, thats an interesting theroy but nearly impossible to implement. 

Your proposed enhancement although it may make building test cases 'easier' it suffers your execution in the long run. Instead of search for an element based xpath once, you are now searching for it multiple times. If you apply this practice to all elements you search for, our test with double or triple in execution time. All to save the user from figuring our how xpath works.
